src/modules/edc-demo/components/contract-agreement-transfer-dialog/contract-agreement-transfer-dialog-form-model.ts

```diff
@@ -11,7 +11,5 @@ export type ContractAgreementTransferDialogFormValue =
  * Form Group Template Type
  */
 export interface ContractAgreementTransferDialogFormModel {
-  dataAddressType: FormControl<DataAddressType | null>;
-  dataDestination: FormControl<string>;
-  httpUrl: FormControl<string>;
+  datasink: FormGroup<ContractAgreementDatasinkFormModel>;
```

Comment:
This is a misunderstanding.

The nested form group should be unnested: Before the fields were here, and they should be here again.

Angular Forms need to be as flat as possible and the nesting of FormGroups into sections "metadata, advanced, datasource" in the asset editor dialog exists only because of mat-steps which requires one form group per step.